### DISCUSSION TITLE
See stream with <video> tag instead of <iframe>

### DISCUSSION THREAD

**Discussion by andreaghezzi** [bluenviron/mediamtx#2926] [Questions]
### Question
I need to see my WebRTC streaming in video tag HTML.  Is it possible to connect to the stream directly from the video tag instead of using iframe? Then do ```<video src="http://localhost:8889/mystreaming/stream1"/>```?

**Comment by asturl** [comment]
I'm not aware of a simpler solution, but you can use the code inside **publish_index.html** in your own frontend as a basis to establish a WebRTC connection via the WHEPClient and a HTML video element of your choice. Some changes are necessary (the WHEP URL will have to be inserted properly with your server/stream scenario) but it works.

**Comment by aler9** [comment]
The \<video> tag doesn't support playing WebRTC streams without an underlying library. The [README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#web-browsers) recommends to either use \<iframe> or to customize publish_index.html.

[a comment by an automated account is not shown]
